airbyte-integrations/bases/source-acceptance-test/unit_tests/test_test_full_refresh.py

```diff
@@ -108,17 +126,109 @@ def record_message_from_record(record_: Dict) -> List[AirbyteMessage]:
-                destination_sync_mode="overwrite",
+recordset_comparison_test_cases = [
+    (
+        [["id"]],
+        [{"id": 1, "first_name": "Thomas", "last_name": "Edison"}, {"id": 2, "first_name": "Nicola", "last_name": "Tesla"}],
+        [{"id": 1, "first_name": "Albert", "last_name": "Einstein"}, {"id": 2, "first_name": "Joseph", "last_name": "Lagrange"}],
+        False,
+        "pk_sets_equal_success",
+    ),
+    (
+        [["id"]],
+        [
+            {"id": 1, "first_name": "Thomas", "last_name": "Edison"},
+        ],
+        [{"id": 1, "first_name": "Albert", "last_name": "Einstein"}, {"id": 2, "first_name": "Joseph", "last_name": "Lagrange"}],
+        False,
+        "pk_first_is_subset_success",
+    ),
+    (
+        [["id"]],
+        [{"id": 1, "first_name": "Thomas", "last_name": "Edison"}, {"id": 2, "first_name": "Nicola", "last_name": "Tesla"}],
+        [{"id": 1, "first_name": "Albert", "last_name": "Einstein"}],
+        True,
+        "pk_second_is_subset_fail",
+    ),
+    (
+        [["id"]],
+        [{"id": 1, "first_name": "Thomas", "last_name": "Edison"}, {"id": 2, "first_name": "Nicola", "last_name": "Tesla"}],
+        [{"id": 2, "first_name": "Thomas", "last_name": "Edison"}, {"id": 3, "first_name": "Nicola", "last_name": "Tesla"}],
+        True,
+        "pk_no_subsets_fail",
+    ),
+    (
+        None,
+        [{"id": 1, "first_name": "Thomas", "last_name": "Edison"}, {"id": 2, "first_name": "Nicola", "last_name": "Tesla"}],
+        [{"id": 1, "first_name": "Thomas", "last_name": "Edison"}, {"id": 2, "first_name": "Nicola", "last_name": "Tesla"}],
+        False,
+        "no_pk_sets_equal_success",
+    ),
+    (
+        None,
+        [
+            {"id": 1, "first_name": "Thomas", "last_name": "Edison"},
+        ],
+        [{"id": 1, "first_name": "Thomas", "last_name": "Edison"}, {"id": 2, "first_name": "Nicola", "last_name": "Tesla"}],
+        False,
+        "no_pk_first_is_subset_success",
+    ),
+    (
+        None,
+        [{"id": 1, "first_name": "Thomas", "last_name": "Edison"}, {"id": 2, "first_name": "Nicola", "last_name": "Tesla"}],
+        [
+            {"id": 1, "first_name": "Thomas", "last_name": "Edison"},
+        ],
+        True,
+        "no_pk_second_is_subset_fail",
+    ),
+    (
+        None,
+        [{"id": 1, "first_name": "Thomas", "last_name": "Edison"}, {"id": 2, "first_name": "Nicola", "last_name": "Tesla"}],
+        [{"id": 2, "first_name": "Nicola", "last_name": "Tesla"}, {"id": 3, "first_name": "Albert", "last_name": "Einstein"}],
+        True,
+        "no_pk_no_subsets_fail",
+    ),
+]
+
+
+@pytest.mark.parametrize(
+    "primary_key, first_read_records, second_read_records, should_fail, test_case_name",
+    recordset_comparison_test_cases,
+    ids=[test_case[-1] for test_case in recordset_comparison_test_cases],
+)
+def test_recordset_comparison(primary_key, first_read_records, second_read_records, should_fail, test_case_name):
```

Comment:
nit
```suggestion
def test_recordset_comparison(mocker, primary_key, first_read_records, second_read_records, should_fail, test_case_name):
```

the `mocker` fixture can be used instead of importing `mock`: `docker_runner_mock = mocker.MagicMock()`